File: Patterns/09. Diamond Patterns/81. Full Diamond Stars/fds.py
n = int(input("Enter a Number : "))

# pyramid
for i in range(1,n+1):
    # print spaces
    for j in range(1,n-i+1):
        print(end='  ')
    # print stars
    for j in range(1,2*i):
        print("*",end=" ")
    print()

#inverted pyramid
for i in range(1,n+1):
    # print spaces
    for j in range(1,i+1):
        print(end='  ')
    # print stars
    for j in range(1,2*(n-i)):
        print("*",end=" ")
    print()


'''
Output:

        * 
      * * * 
    * * * * * 
  * * * * * * * 
* * * * * * * * * 
  * * * * * * * 
    * * * * * 
      * * * 
        * 

        
Another Method using if-else

row=1

for i in range(1,2*n):
    if(i<=n):
        row=i
    else:
        row=2*n-i
    
    # print spaces
    for j in range(1,n-row+1):
        print(end='  ')
    # print stars
    for j in range(1,2*row):
        print("*", end=' ')
    print()

'''

File: Patterns/9. Diamond Patterns/81. Full Diamond Stars/fds.cpp
#include <iostream>
using namespace std;
int main(){
    int n;
    cout<<"Enter a Number : ";
    cin>>n;

    // pyramid
    for(int i=1; i<=n; i++){
        // print spaces
        for(int j=1; j<=n-i; j++){
            cout<<"  ";
        }
        // print stars
        for(int j=1; j<=2*i-1; j++){
            cout<<"* ";
        }
        cout<<'\n';
    }

    // Inverted Pyramid
    for(int i=1; i<n; i++){
        // print spaces
        for(int j=1; j<=i; j++){
            cout<<"  ";
        }
        // print stars
        for(int j=1; j<=2*(n-i)-1; j++){
            cout<<"* ";
        }
        cout<<'\n';
    }
}

/*
Output

        * 
      * * * 
    * * * * * 
  * * * * * * * 
* * * * * * * * * 
  * * * * * * * 
    * * * * * 
      * * * 
        *  

Another Method using if-else

    int row;

    for(int i=1; i<=2*n-1; i++){
        if(i<=n){
            row = i;
        }else{
            row = 2*n-i;
        }
        // print spaces
        for(int j=1; j<=n-row; j++){
            cout<<"  ";
        }
        // print stars
        for(int j=1; j<=2*row-1; j++){
            cout<<"* ";
        }
        cout<<'\n';

    }

*/